Mirror/Runtime/NetworkTransformBase.cs

```diff
@@ -0,0 +1,545 @@
+        {
+            // note: C# byte - byte => int, hence so many casts
+            int targetRange = maxTarget - minTarget; // max byte - min byte only fits into something bigger
+            float valueRange = maxValue - minValue;
+            float valueRelative = value - minValue;
+            return (byte)(minTarget + (byte)(valueRelative/valueRange * (float)targetRange));
+        }
+
+        // ScaleUShortToFloat(    0, ushort.MinValue, ushort.MaxValue, -1, 1) => -1
+        // ScaleUShortToFloat(32767, ushort.MinValue, ushort.MaxValue, -1, 1) => 0
+        // ScaleUShortToFloat(49151, ushort.MinValue, ushort.MaxValue, -1, 1) => 0.4999924
+        // ScaleUShortToFloat(65535, ushort.MinValue, ushort.MaxValue, -1, 1) => 1
+        static float ScaleUShortToFloat(ushort value, ushort minValue, ushort maxValue, float minTarget, float maxTarget)
+        {
+            // note: C# ushort - ushort => int, hence so many casts
+            float targetRange = maxTarget - minTarget;
+            ushort valueRange = (ushort)(maxValue - minValue);
+            ushort valueRelative = (ushort)(value - minValue);
+            return minTarget + (float)((float)valueRelative/(float)valueRange * targetRange);
+        }
+
+        // ScaleByteToFloat(  0, byte.MinValue, byte.MaxValue, -1, 1) => -1
+        // ScaleByteToFloat(127, byte.MinValue, byte.MaxValue, -1, 1) => -0.003921569
+        // ScaleByteToFloat(191, byte.MinValue, byte.MaxValue, -1, 1) => 0.4980392
+        // ScaleByteToFloat(255, byte.MinValue, byte.MaxValue, -1, 1) => 1
+        static float ScaleByteToFloat(byte value, byte minValue, byte maxValue, float minTarget, float maxTarget)
+        {
+            // note: C# byte - byte => int, hence so many casts
+            float targetRange = maxTarget - minTarget;
+            byte valueRange = (byte)(maxValue - minValue);
+            byte valueRelative = (byte)(value - minValue);
+            return minTarget + ((float)valueRelative/(float)valueRange * targetRange);
+        }
+
+        // eulerAngles have 3 floats, putting them into 2 bytes of [x,y],[z,0]
+        // would be a waste. instead we compress into 5 bits each => 15 bits.
+        // so a ushort.
+        static ushort PackThreeFloatsIntoUShort(float u, float v, float w, float minValue, float maxValue)
+        {
+            // 5 bits max value = 1+2+4+8+16 = 31 = 0x1F
+            byte lower = ScaleFloatToByte(u, minValue, maxValue, 0x00, 0x1F);
+            byte middle = ScaleFloatToByte(v, minValue, maxValue, 0x00, 0x1F);
+            byte upper = ScaleFloatToByte(w, minValue, maxValue, 0x00, 0x1F);
+            ushort combined = (ushort)(upper << 10 | middle << 5 | lower);
+            return combined;
+        }
+
+        // see PackThreeFloatsIntoUShort for explanation
+        static float[] UnpackUShortIntoThreeFloats(ushort combined, float minTarget, float maxTarget)
```

Comment:
Return Vector3 instead,   it is much faster and it is type safe because you know exactly how many values you will get.

also,  unit test